tests/gocase/unit/type/set/set_test.go

```diff
@@ -0,0 +1,741 @@
+		CreateSet(t, rdb, ctx, "myset", []interface{}{"a", "b", "c", "d"})
+		require.EqualValues(t, 0, rdb.SRem(ctx, "myset", "k", "k", "k").Val())
+		require.EqualValues(t, 2, rdb.SRem(ctx, "myset", "b", "d", "x", "y").Val())
+		cmd := rdb.SMembers(ctx, "myset")
+		require.NoError(t, cmd.Err())
+		sort.Strings(cmd.Val())
+		require.EqualValues(t, []string{"a", "c"}, cmd.Val())
+	})
+
+	t.Run("SREM variadic version with more args needed to destroy the key", func(t *testing.T) {
+		CreateSet(t, rdb, ctx, "myset", []interface{}{1, 2, 3})
+		require.EqualValues(t, 3, rdb.SRem(ctx, "myset", 1, 2, 3, 4, 5, 6, 7, 8).Val())
+	})
+
+	t.Run("SREM variadic version with more args needed to destroy the key", func(t *testing.T) {
+		CreateSet(t, rdb, ctx, "myset", []interface{}{1, 2, 3})
+		require.EqualValues(t, 3, rdb.SRem(ctx, "myset", 1, 2, 3, 4, 5, 6, 7, 8).Val())
+	})
+
+	for _, dstype := range [2]string{"hashtable", "intset"} {
+		for i := 1; i <= 5; i++ {
+			require.NoError(t, rdb.Del(ctx, "set"+strconv.Itoa(i)).Err())
+		}
+		for i := 0; i < 200; i++ {
+			require.NoError(t, rdb.SAdd(ctx, "set1", i).Err())
+			require.NoError(t, rdb.SAdd(ctx, "set2", i+195).Err())
+		}
+		for _, i := range []int{199, 195, 100, 2000} {
+			require.NoError(t, rdb.SAdd(ctx, "set3", i).Err())
+		}
+		for i := 5; i < 200; i++ {
+			require.NoError(t, rdb.SAdd(ctx, "set4", i).Err())
+		}
+		var larger interface{}
+		larger = 200
+		require.NoError(t, rdb.SAdd(ctx, "set5", 0).Err())
+		if dstype == "hashtable" {
+			larger = "foo"
+		}
+		for i := 1; i <= 5; i++ {
+			switch larger := larger.(type) {
+			default:
+				require.NoError(t, rdb.SAdd(ctx, "set"+strconv.Itoa(i), larger).Err())
+			}
+		}
+		t.Run("SINTER with two sets - "+dstype, func(t *testing.T) {
+			cmd := rdb.SInter(ctx, "set1", "set2")
+			require.NoError(t, cmd.Err())
+			sort.Strings(cmd.Val())
+			switch larger := larger.(type) {
+			case int:
+				require.EqualValues(t, []string{"195", "196", "197", "198", "199", strconv.Itoa(larger)}, cmd.Val())
+			case string:
+				require.EqualValues(t, []string{"195", "196", "197", "198", "199", larger}, cmd.Val())
+			}
+		})
+
+		t.Run("SINTERSTORE with two sets - "+dstype, func(t *testing.T) {
+			require.NoError(t, rdb.SInterStore(ctx, "setres", "set1", "set2").Err())
+			cmd := rdb.SMembers(ctx, "setres")
+			require.NoError(t, cmd.Err())
+			sort.Strings(cmd.Val())
+
+			switch larger := larger.(type) {
+			case int:
+				require.EqualValues(t, []string{"195", "196", "197", "198", "199", strconv.Itoa(larger)}, cmd.Val())
+			case string:
+				require.EqualValues(t, []string{"195", "196", "197", "198", "199", larger}, cmd.Val())
+			}
+		})
+
+		t.Run("SUNION with two sets - "+dstype, func(t *testing.T) {
+			set1 := rdb.SMembers(ctx, "set1")
+			require.NoError(t, set1.Err())
+			set2 := rdb.SMembers(ctx, "set2")
+			require.NoError(t, set2.Err())
+			var expect []string
+			var vis = make(map[string]bool)
+			for _, i := range set1.Val() {
```

Comment:
I saw this unique merge code block appears many times, do you mind adding a util function for merging N sets?

Reply:
ok. I have added it.

Reply:
Thanks